### ISSUE TITLE
Silencing the REPL

### ISSUE THREAD

**Issue by jochemstoel** [adamralph/dude#9]
Could you perhaps explain to me where and what I need to modify in your source code to _silence_ the dude REPL stdout of evaluator context unrelated strings?  In other words, I would like to hide the version message and the *>* symbol that is prepended to each line so that only the actual evaluator is left.

So the *>* symbol before each new line, and this:
> Microsoft (R) Visual C# Interactive Compiler version 1.0.0.0
> Copyright (C) Microsoft Corporation. All rights reserved.

> Type "#help" for more information.

I found a bunch of strings called _LogoLine1_ and _LogoLine2_ in some XML files but I can't find where they are actually being written to the stdout. If I search the folder for the string _LogoLine1_ then there is nowhere such a thing like `Console.Write(LogoLine1)`

Thanks

**Comment by adamralph** [comment]
@jochemstoel thanks for raising this.

The console output is entirely handled by the Roslyn packages. Dude calls into https://github.com/dotnet/roslyn/blob/311e9475f7e80226ee4c4586abc5d5a9d51e8683/src/Scripting/Core/Hosting/CommandLine/CommandLineRunner.cs#L46 and the output comes from somewhere inside that assembly or its dependencies.

Dude is a very simple wrapper around those libraries, just like csi.exe. Essentially, each of these apps is a single file:

- csi.exe: https://github.com/dotnet/roslyn/blob/e045283767da82cfd276d020c8a798f78513a1ab/src/Interactive/CsiCore/Csi.cs
- Dude: https://github.com/adamralph/dude/blob/f3281bb698a407760d13261cc34592095b7b4b9a/src/Dude.Console/Program.cs

The Dude code is effectively identical to that in csi.exe, but all the calls to the Roslyn types are made via reflection since they are all internal (but visible to csi.exe).

So, in summary, there is no way to change this in Dude. The change must be made in csi.exe, over in https://github.com/dotnet/roslyn. Perhaps the Roslyn team are willing to add a command line switch to suppress the unwanted output? If so, as soon as csi.exe supports this, and the new versions of the underlying NuGet packages (e.g. Microsoft.CodeAnalysis.Scripting.Hosting) have been published, then Dude can be updated to use those new versions and it will match the behaviour of the new csi.exe.

**Comment by jochemstoel** [comment]
Thanks for taking the time to answer. You are right about the dependencies. I should have looked better. Caught a glimpse of the source code of dude and I thought you created a lot more yourself. I was searching for a sharp REPL hoping one of them would include some sort of programmatically accessible evaluator component that I can isolate.  Like Mono.CSharp but more developed.

**Comment by adamralph** [comment]
@jochemstoel what is your end goal? Do you want to evaluate some arbitrary C# snippet? If so, you can use the `Microsoft.CodeAnalysis.CSharp.Scripting` NuGet package and write something like:

```C#
using Microsoft.CodeAnalysis.CSharp.Scripting;
// ...
var result = await CSharpScript.EvaluateAsync<int>("1 + 2");
```

**Comment by jochemstoel** [comment]
I don't have an end goal I already reached it. I am looking into replacing
the Mono csharp evaluator in my client monitoring services with something
better if it exists because I'm not gonna invent it myself.

**Comment by adamralph** [comment]
@jochemstoel it sounds like you should indeed be using the `Microsoft.CodeAnalysis.CSharp.Scripting` package, as I described above.

**Comment by adamralph** [comment]
@jochemstoel if you have more questions on this thread, please re-open the issue.
